tests/test_meta_functions_against_registry.py

```diff
@@ -9,6 +10,159 @@
+            else:
+                # both should revert for consistency:
+                assert metaregistry_revert and actual_output_revert
+
+
+def test_get_underlying_decimals(metaregistry, registries, max_pools, curve_api):
+    def get_coin_decimals_from_pool(pool):
+        coin_decimals = []
+        for i in range(4):
+            try:
+                coin_decimals.append(curve_api.get_decimals_for_coin_in_pool(pool, i))
+            except brownie.exceptions.VirtualMachineError:
+                break
+        coin_decimals += [0] * (8 - len(coin_decimals))
+        return tuple(coin_decimals)
+
+    for i, registry in enumerate(registries):
+
+        total_pools = registry.pool_count() if max_pools == 0 else max_pools
+        for pool_index in range(total_pools):
+            pool = registry.pool_list(pool_index)
+            metaregistry_output = metaregistry.get_underlying_decimals(pool)
+
+            if (
+                i == METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX
+                or i == METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX
+            ):
+                try:
+                    actual_output = registry.get_underlying_decimals(pool)
+                    if not actual_output:
+                        continue
+                    assert actual_output == metaregistry_output
+                except AssertionError:
+                    print(f"metaregistry_output: {metaregistry_output}")
+                    print(f"child registry output: {actual_output}")
+                    print(
+                        "registry and metaregistry don't match. "
+                        "Checking if MetaRegistry is truly at fault here..."
+                    )
+                    # If a pool is in the stable factory, the underlying_decimals
+                    # method in the factory or registry only works correctly
+                    # for metapools. Otherwise, it returns a 0 if it isn't paired with
+                    # a basepool. So, get the correct coin decimals in case a factory
+                    # pool is not a metapool:
+                    actual_output = get_coin_decimals_from_pool(pool)
+                    assert actual_output == metaregistry_output
+
+            else:
+                # Crypto Registries on mainnet do not have get_underlying_decimals,
+                # so check each coin's decimals with Metaregistry's output:
+                actual_output = get_coin_decimals_from_pool(pool)
+                assert actual_output == metaregistry_output
+
+
+def test_get_underlying_coins(metaregistry, registries, max_pools, curve_api):
+    for i, registry in enumerate(registries):
+        total_pools = registry.pool_count() if max_pools == 0 else max_pools
+        for pool_index in range(total_pools):
+            metaregistry_output = None
+            pool = registry.pool_list(pool_index)
+            attempt = 0
+            while attempt <= 10:
```

Comment:
What is the purpose of this?  Confusing, if there's a good reason than worth documenting

Reply:
Removed.